packages/headless/src/features/folding/folding-slice.test.ts

```diff
@@ -1,163 +1,144 @@
+  collectionName: string,
+  root: Hierarchy,
+  parentName?: string
+): ResultWithFolding[] {
+  const result = buildMockResultWithFolding({
+    uniqueId: root.name,
+    title: root.name,
     raw: {
       urihash: '',
-      collection,
-      parent: 'animals',
+      collection: collectionName,
+      id: root.name,
     },
-    parentResult: {...animalsResult},
   });
 
-  const relevantResult = buildMockResultWithFolding({
-    uniqueId: '_reptiles',
-    raw: {
-      urihash: '',
-      collection,
-      id: ['reptiles'],
-      parent: 'animals',
-    },
-    parentResult: {...animalsResult},
-    childResults: [mammalsResult, ballPythonsResult, geckosResult],
-  });
+  if (parentName) {
+    result.raw.parent = parentName;
+  }
 
-  return {
-    animalsResult,
-    relevantResult,
-    ballPythonsResult,
-    geckosResult,
-    mammalsResult,
-  };
+  return [
+    result,
+    ...(root.children ?? []).flatMap((child) =>
+      buildMockResultsFromHierarchy(collectionName, child, root.name)
+    ),
+  ];
 }
 
-describe('folding slice', () => {
-  it('should be disabled by default', () => {
-    expect(foldingReducer(undefined, {type: ''}).enabled).toBeFalsy();
-  });
+function emulateAPIFolding(
+  results: ResultWithFolding[],
+  target: ResultWithFolding
+): ResultWithFolding {
+  const newResult: ResultWithFolding = {...target};
 
-  it('should correctly fold hierarchical results', () => {
-    const {
-      animalsResult,
-      relevantResult,
-      ballPythonsResult,
-      geckosResult,
-      mammalsResult,
-    } = getEncyclopediaCollection();
-
-    const otherResult = buildMockResultWithFolding();
-
-    const initialState: FoldingState = {
-      ...getFoldingInitialState(),
-      enabled: true,
-      fields: {collection: 'collection', parent: 'parent', child: 'id'},
-    };
+  newResult.childResults = results
+    .filter((result) => result.raw.parent === target.raw.id)
+    .map((result) => emulateAPIFolding(results, result));
 
-    const action = executeSearch.fulfilled(
-      buildMockSearch({
-        response: buildMockSearchResponse({
-          results: [relevantResult, otherResult],
-        }),
-      }),
-      '',
-      null as never
-    );
-
-    const expectedHierarchy = {
-      ...animalsResult,
-      children: [
-        {
-          ...relevantResult,
-          children: [
-            {...ballPythonsResult, children: []},
-            {...geckosResult, children: []},
-          ],
-        },
-        {...mammalsResult, children: []},
-      ],
+  if (target.raw.parent) {
+    newResult.parentResult = {
+      ...results.find((result) => result.raw.id === target.raw.parent)!,
     };
+    newResult.childResults.push({...newResult.parentResult});
+  }
 
-    const expectedOtherResult = {...otherResult, children: []};
+  return newResult;
+}
 
-    const results = foldingReducer(initialState, action).collections;
+function extractHierarchy(root: FoldedResult): Hierarchy {
```

Comment:
This util takes a result folded by Headless and resolves what hierarchy it's equivalent to. It's only meant to test whether results were folded correctly. Without it, testing if results were folded correctly is a mess.